src/scripts/main.js

```diff
@@ -2,4 +2,33 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+  const wallRectangle = wall.getBoundingClientRect();
+
+  let x = e.clientX - wallRectangle.left - spider.width + spider.width / 2;
```

Comment:
The calculation for `x` should ensure that the spider does not exceed the left boundary of the wall. The current calculation might cause the spider to exceed the left boundary if `e.clientX` is less than `wallRectangle.left + spider.width / 2`.